pipeline/backend/kubernetes/image_pull_secret.go

```diff
@@ -0,0 +1,57 @@
+package kubernetes
+
+import (
+	"encoding/json"
+
+	"github.com/distribution/reference"
+	config_file "github.com/docker/cli/cli/config/configfile"
+	config_file_types "github.com/docker/cli/cli/config/types"
+	"go.woodpecker-ci.org/woodpecker/v2/pipeline/backend/types"
+	"go.woodpecker-ci.org/woodpecker/v2/pipeline/frontend/yaml/utils"
+	v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func needsImagePullSecret(step *types.Step) bool {
+	return step.AuthConfig.Username != "" && step.AuthConfig.Password != ""
+}
+
+func mkImagePullSecret(step *types.Step, config *config, podName, goos string, options BackendOptions) (*v1.Secret, error) {
```

Comment:
Unused `goos`.

Reply:
There is `secrets.go`. Secrets stuff should be there, I think.

Reply:
Removed goos, moved all secrets-related func's to secrets.go